crates/sui/src/upgrade_compatibility.rs

```diff
@@ -813,6 +821,47 @@ fn module_compatibility_error_diag(
     Ok(diags)
 }
 
+const PACKAGE_TABLE: &str = "[package]";
+fn missing_module_diag(
+    module_name: &Identifier,
+    package_path: &PathBuf,
+) -> Result<Diagnostics, Error> {
+    let mut diags = Diagnostics::new();
+
+    // read Move.toml to get the hash and first line start and end
+    let toml_path = package_path.join("Move.toml");
+    let toml_str = fs::read_to_string(&toml_path).context("Unable to read Move.toml")?;
+    let hash = FileHash::new(&toml_str);
+
+    let start: usize = toml_str
+        .find(PACKAGE_TABLE)
+        .context("Malformed Move.toml")?;
```

Comment:
It looks like if we fail to read the `Move.toml` we may end up hiding all the other diagnostics. Do you think it would be possible for us to always produce some kind of `Diagnostics` value, even if we failed to  read the `Move.toml` to avoid that?

Reply:
I was thinking this more as a "critical" error however I think it could be good to just let this return a diagnostic for byte ranges `0,0` as a fallback. Some errors I'd guess will just fall in that "critical" category and should be errored out but we could potentially discuss follow up work to prevent errors from stopping execution by converting `Errors -> Diags`.   Or alternatively just maintain a list of errors and print them last / concatenate them into a single error that is thrown last.